chain/chain/src/chain.rs

```diff
@@ -2218,6 +2219,17 @@ impl Chain {
             panic!("The client protocol version is older than the protocol version of the network. Please update nearcore. Client protocol version:{}, network protocol version {}", PROTOCOL_VERSION, epoch_protocol_version);
         }
 
+        let next_epoch_id = self.epoch_manager.get_next_epoch_id_from_prev_block(prev_hash)?;
+        let next_epoch_protocol_version =
+            self.epoch_manager.get_epoch_protocol_version(&next_epoch_id)?;
+        if next_epoch_protocol_version > PROTOCOL_VERSION {
+            error!(
+                "The client protocol version is about to be superseded, please upgrade nearcore as soon as possible. Client protocol version {}, new network protocol version {}",
+                PROTOCOL_VERSION,
+                next_epoch_protocol_version,
+            );
+        }
+
```

Comment:
nit1 - Can you move that to a helper method? This one is too long as is. 
nit2 - I typically prefer to perform checks in postprocess block - it just feels a bit safer knowing that the block is valid and processed.

Reply:
done